### PR TITLE
document api/core and api/key packages

### PR DESCRIPTION
**Pull request by krnowak**

### CODE REVIEW

File: api/core/number.go [hunk 1]
```diff
@@ -94,14 +94,14 @@ func (n Number) AsUint64() uint64 {
 
 // - as x atomic
 
-// AsNumberAtomic gets the raw, uninterpreted raw value. Might be useful for
-// some atomic operations.
+// AsNumberAtomic atomically gets the raw, uninterpreted raw
+// value. Might be useful for some atomic operations.
```
Comment (rghetia) [inline api/core/number.go]: maybe rephrase it to `AsNumberAtomic gets uninterpreted raw Number atomically. Might be useful for some atomic operations`

File: api/core/number.go [hunk 2]
```diff
@@ -94,14 +94,14 @@ func (n Number) AsUint64() uint64 {
...
 func (n *Number) AsNumberAtomic() Number {
 	return NewNumberFromRaw(n.AsRawAtomic())
 }
 
-// AsRawAtomic gets atomically the raw, uninterpreted raw value. Might
-// be useful for some atomic operations.
+// AsRawAtomic atomically gets atomically the raw, uninterpreted raw
+// value. Might be useful for some atomic operations.
```
Comment (rghetia) [inline api/core/number.go]: Similarly, 'AsNumberAtomic gets uninterpreted raw value atomically. Might be useful for some atomic operations'

File: api/core/number.go [hunk 3]
```diff
@@ -113,12 +113,12 @@ func (n *Number) AsInt64Atomic() int64 {
 }
 
 // AsFloat64 assumes that the measurement value contains a float64 and
-// returns it as such.
+// atomically returns it as such.
```
Comment (rghetia) [inline api/core/number.go]: s/AsFloat64/AsFloat64Atomic/

File: api/core/number.go [hunk 4]
```diff
@@ -190,74 +202,106 @@ func (n Number) CoerceToUint64(kind NumberKind) uint64 {
...
+// SetUint64 assumes that the number contains a uint64 and sets it to
+// the passed value.
 func (n *Number) SetUint64(u uint64) {
 	*n.AsUint64Ptr() = u
 }
 
 // - set atomic
 
+// SetNumberAtomic atomically sets the number to the passed
+// number. Both should be of the same kind.
```
Comment (rghetia) [inline api/core/number.go]: maybe rephrase it `SetNumberAtomic sets the number to the passed number atomically. Both should be of the same kind.`

File: api/core/number.go [hunk 5]
```diff
@@ -190,74 +202,106 @@ func (n Number) CoerceToUint64(kind NumberKind) uint64 {
...
 func (n *Number) SetNumberAtomic(nn Number) {
 	atomic.StoreUint64(n.AsRawPtr(), nn.AsRaw())
 }
 
+// SetRawAtomic atomically sets the number to some raw value. Both number
+// and the raw number should represent the same kind.
```
Comment (rghetia) [inline api/core/number.go]: same.

File: api/core/number.go (hunk 1 above)
Comment (krnowak) [reply api/core/number.go]: Naah, I need to put more "raw" words there. Two times is not enough. ;)

Anyway, thanks for the suggestion. And that sentence about usefulness for atomic ops is a wrong copy pasta, so I dropped it.

File: api/core/number.go (hunk 2 above)
Comment (krnowak) [reply api/core/number.go]: Thanks, fixed.

File: api/core/number.go (hunk 3 above)
Comment (krnowak) [reply api/core/number.go]: Oops, another copy-pasta turned bad. Fixed.

File: api/core/number.go (hunk 4 above)
Comment (krnowak) [reply api/core/number.go]: Thanks. Fixed.

File: api/core/number.go (hunk 5 above)
Comment (krnowak) [reply api/core/number.go]: Fixed.
